src/sources.jl

```diff
@@ -25,25 +25,34 @@ struct UniformPort <: BoundaryCondition
   ε :: Float64
 end
 
-struct SpeciesSource
-  species :: AbstractSpecies
+abstract type SpeciesSource end
+abstract type SpeciesLoader end
+
+struct ParticleSource <: SpeciesSource
+  species :: Particles
   rate :: TemporalFunction
   x :: PositionDistribution
   v :: VelocityDistribution
   drift :: Vector{Pair{Symbol, Float64}}
 end
 
-struct SpeciesLoader
-  species :: AbstractSpecies
-  density :: Float64
+struct ParticleLoader{T} <: SpeciesLoader
+  species :: Particles
+  value :: T
```

Comment:
I suggest `count` as an Integer64. If a user wants to prescribe number density they have to calculate the corresponding number of particles (assuming they know the volume)